### ISSUE TITLE
build errors with 3.1.72.0

### ISSUE THREAD

**Issue by graysky2** [snort3/snort3#325]
I am getting errors when building using OpenWRT's build system.  The `Makefile` can be found [here](https://github.com/openwrt/packages/blob/master/net/snort3/Makefile).  Any suggestions?  Upstream bug?
```
-- The CXX compiler identification is GNU 12.3.0
-- The C compiler identification is GNU 12.3.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /scratch/union/staging_dir/toolchain-x86_64_gcc-12.3.0_musl/bin/x86_64-openwrt-linux-musl-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /scratch/union/staging_dir/toolchain-x86_64_gcc-12.3.0_musl/bin/x86_64-openwrt-linux-musl-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Found PkgConfig: /scratch/union/staging_dir/host/bin/pkg-config (found version "1.9.5") 
-- Checking for module 'libdaq>=3.0.12'
--   Package 'libdaq', required by 'virtual:world', not found
-- Found DAQ: /scratch/union/staging_dir/target-x86_64_musl/usr/lib/daq3/libdaq.so  
-- Found DNET: /scratch/union/staging_dir/target-x86_64_musl/usr/include  
-- Found FLEX: /scratch/union/staging_dir/host/bin/flex (found suitable version "2.6.4", minimum required is "2.6.0") 
-- Checking for module 'hwloc'
--   Found hwloc, version 2.8.0
-- Found HWLOC: /scratch/union/staging_dir/target-x86_64_musl/usr/lib/libhwloc.so  
-- Checking for module 'luajit'
--   Found luajit, version 2.1.0-beta3
-- Found LuaJIT: /scratch/union/staging_dir/target-x86_64_musl/usr/lib/libluajit-5.1.so (found version "2.1.0-beta3") 
-- Found OpenSSL: /scratch/union/staging_dir/target-x86_64_musl/usr/lib/libcrypto.so (found suitable version "3.0.11", minimum required is "1.1.1")  
-- Found PCAP: /scratch/union/staging_dir/target-x86_64_musl/usr/lib/libpcap.so  
-- Performing Test PCAP_LINKS_SOLO
-- Performing Test PCAP_LINKS_SOLO - Success
-- Checking for module 'libpcre'
--   Found libpcre, version 8.45
-- Found PCRE: /scratch/union/staging_dir/target-x86_64_musl/usr/include  
-- Checking for one of the modules 'zlib'
-- Found ZLIB: /scratch/union/staging_dir/target-x86_64_musl/usr/lib/libz.so (found version "1.3.#define ZLIB_VERSION "1.3"")  
-- Checking for one of the modules 'liblzma'
CMake Warning (dev) at /scratch/union/staging_dir/host/share/cmake-3.26/Modules/FindPackageHandleStandardArgs.cmake:438 (message):
  The package name passed to `find_package_handle_standard_args` (W3M) does
  not match the name of the calling package (Asciidoc).  This can lead to
  problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  cmake/FindAsciidoc.cmake:34 (find_package_handle_standard_args)
  cmake/include_libraries.cmake:20 (find_package)
  CMakeLists.txt:27 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- install w3m to build all-in-one text user manual (missing: W3M_EXE) 
-- Checking for module 'libhs'
--   Found libhs, version 5.4.2
-- Checking for module 'libsafec'
--   Package 'libsafec', required by 'virtual:world', not found
-- Performing Test ICONV_IS_BUILT_IN
-- Performing Test ICONV_IS_BUILT_IN - Success
-- Checking for module 'uuid'
--   Found uuid, version 2.39.2
-- Checking for module 'libunwind'
--   Found libunwind, version 1.6.2
-- Found Libunwind: /scratch/union/staging_dir/target-x86_64_musl/usr/lib/libunwind.so (found version "1.6.2") 
-- Checking for module 'libnuma'
--   Package 'libnuma', required by 'virtual:world', not found
-- Performing Test HAS_C_HIDDEN
-- Performing Test HAS_C_HIDDEN - Success
-- Performing Test HAS_CXX_HIDDEN
-- Performing Test HAS_CXX_HIDDEN - Success
-- Checking for module 'libtcmalloc'
--   Package 'libtcmalloc', required by 'virtual:world', not found
-- Found TCMalloc: /scratch/union/staging_dir/target-x86_64_musl/usr/lib/libtcmalloc.so (found version "") 
-- Looking for malloc_trim
-- Looking for malloc_trim - not found
-- Looking for memrchr
-- Looking for memrchr - found
-- Looking for sigaction
-- Looking for sigaction - found
-- Looking for basename_r
-- Looking for basename_r - not found
-- Performing Test HAVE_GNU_STRERROR_R
-- Performing Test HAVE_GNU_STRERROR_R - Failed
-- Looking for getrpcent
-- Looking for getrpcent - found
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of long int
-- Check size of long int - done
-- Check size of unsigned long int
-- Check size of unsigned long int - done
-- Performing Test INLINE
-- Performing Test INLINE - Success
-- Performing Test RESTRICT
-- Performing Test RESTRICT - Success
-- Looking for hs_scan in /scratch/union/staging_dir/target-x86_64_musl/usr/lib/libhs.so
-- Looking for hs_scan in /scratch/union/staging_dir/target-x86_64_musl/usr/lib/libhs.so - found
-- Looking for uuid_parse in /scratch/union/staging_dir/target-x86_64_musl/usr/lib/libuuid.so
-- Looking for uuid_parse in /scratch/union/staging_dir/target-x86_64_musl/usr/lib/libuuid.so - found

-------------------------------------------------------
snort version 3.1.72.0

Install options:
    prefix:     /usr
    includes:   /usr/include/snort
    plugins:    /usr/lib/snort

Compiler options:
    CC:             /scratch/union/staging_dir/toolchain-x86_64_gcc-12.3.0_musl/bin/x86_64-openwrt-linux-musl-gcc
    CXX:            /scratch/union/staging_dir/toolchain-x86_64_gcc-12.3.0_musl/bin/x86_64-openwrt-linux-musl-g++
    CFLAGS:         -march=znver3 -mtune=znver3 -O2 -pipe -fno-caller-saves -fno-plt -fhonour-copts -fmacro-prefix-map=/scratch/union/build_dir/target-x86_64_musl/snort3-3.1.72.0=snort3-3.1.72.0 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -I/scratch/union/staging_dir/target-x86_64_musl/usr/include/daq3 -I/scratch/union/staging_dir/target-x86_64_musl/usr/include/tirpc  -fvisibility=hidden   -DNDEBUG  -fno-builtin-malloc -fno-builtin-calloc -fno-builtin-realloc -fno-builtin-free  -DNDEBUG
    CXXFLAGS:       -march=znver3 -mtune=znver3 -O2 -pipe -fno-caller-saves -fno-plt -fhonour-copts -fmacro-prefix-map=/scratch/union/build_dir/target-x86_64_musl/snort3-3.1.72.0=snort3-3.1.72.0 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -I/scratch/union/staging_dir/target-x86_64_musl/usr/include/daq3 -I/scratch/union/staging_dir/target-x86_64_musl/usr/include/tirpc  -fvisibility=hidden   -DNDEBUG  -fno-builtin-malloc -fno-builtin-calloc -fno-builtin-realloc -fno-builtin-free  -DNDEBUG
    EXE_LDFLAGS:    -L/scratch/union/staging_dir/toolchain-x86_64_gcc-12.3.0_musl/usr/lib -L/scratch/union/staging_dir/toolchain-x86_64_gcc-12.3.0_musl/lib -fuse-ld=bfd -znow -zrelro -L/scratch/union/staging_dir/target-x86_64_musl/usr/lib/daq3 -ltirpc    
    MODULE_LDFLAGS: -L/scratch/union/staging_dir/toolchain-x86_64_gcc-12.3.0_musl/usr/lib -L/scratch/union/staging_dir/toolchain-x86_64_gcc-12.3.0_musl/lib -fuse-ld=bfd -znow -zrelro -L/scratch/union/staging_dir/target-x86_64_musl/usr/lib/daq3 -ltirpc -Wl,-Bsymbolic-functions    

Feature options:
    DAQ Modules:    Dynamic
    libatomic:      User-specified
    Hyperscan:      ON
    ICONV:          ON
    Libunwind:      OFF
    LZMA:           OFF
    RPC DB:         TIRPC
    SafeC:          OFF
    TCMalloc:       ON
    JEMalloc:       OFF
    UUID:           ON
    NUMA:           OFF
-------------------------------------------------------

-- Configuring done (1.6s)
-- Generating done (0.1s)
-- Build files have been written to: /scratch/union/build_dir/target-x86_64_musl/snort3-3.1.72.0
touch /scratch/union/build_dir/target-x86_64_musl/snort3-3.1.72.0/.configured_68b329da9893e34099c7d8ad5cb9c940
rm -f /scratch/union/build_dir/target-x86_64_musl/snort3-3.1.72.0/.built
touch /scratch/union/build_dir/target-x86_64_musl/snort3-3.1.72.0/.built_check
MAKEFLAGS="--jobserver-auth=fifo:/tmp/GMfifo3582521" /scratch/union/staging_dir/host/bin/ninja   -C /scratch/union/build_dir/target-x86_64_musl/snort3-3.1.72.0 
ninja: Entering directory `/scratch/union/build_dir/target-x86_64_musl/snort3-3.1.72.0'
[66/986] Building CXX object src/detection/CMakeFiles/detection.dir/fp_create.cc.o
FAILED: src/detection/CMakeFiles/detection.dir/fp_create.cc.o 
/scratch/union/staging_dir/toolchain-x86_64_gcc-12.3.0_musl/bin/x86_64-openwrt-linux-musl-g++ -DHAVE_CONFIG_H -Dinline=inline -Drestrict=__restrict -I/scratch/union/build_dir/target-x86_64_musl/snort3-3.1.72.0/src/network_inspectors -I/scratch/union/build_dir/target-x86_64_musl/snort3-3.1.72.0/src -I/scratch/union/staging_dir/target-x86_64_musl/usr/include/luajit-2.1 -I/scratch/union/build_dir/target-x86_64_musl/snort3-3.1.72.0 -I/scratch/union/staging_dir/host/include -I/scratch/union/staging_dir/target-x86_64_musl/usr/include/hs -I/scratch/union/staging_dir/target-x86_64_musl/usr/include/uuid -march=znver3 -mtune=znver3 -O2 -pipe -fno-caller-saves -fno-plt -fhonour-copts -fmacro-prefix-map=/scratch/union/build_dir/target-x86_64_musl/snort3-3.1.72.0=snort3-3.1.72.0 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -I/scratch/union/staging_dir/target-x86_64_musl/usr/include/daq3 -I/scratch/union/staging_dir/target-x86_64_musl/usr/include/tirpc  -fvisibility=hidden   -DNDEBUG  -fno-builtin-malloc -fno-builtin-calloc -fno-builtin-realloc -fno-builtin-free  -DNDEBUG -std=c++14 -MD -MT src/detection/CMakeFiles/detection.dir/fp_create.cc.o -MF src/detection/CMakeFiles/detection.dir/fp_create.cc.o.d -o src/detection/CMakeFiles/detection.dir/fp_create.cc.o -c /scratch/union/build_dir/target-x86_64_musl/snort3-3.1.72.0/src/detection/fp_create.cc
In file included from /scratch/union/build_dir/target-x86_64_musl/snort3-3.1.72.0/src/detection/fp_create.cc:41:
/scratch/union/build_dir/target-x86_64_musl/snort3-3.1.72.0/src/hash/xhash.h:63:39: error: 'u_int8_t' has not been declared
   63 |     int release_node(const void* key, u_int8_t type = 0);
      |                                       ^~~~~~~~
[70/986] Building CXX object src/detection/CMakeFiles/detection.dir/detection_module.cc.o
/scratch/union/build_dir/target-x86_64_musl/snort3-3.1.72.0/src/detection/detection_module.cc: In member function 'virtual bool snort::DetectionModule::set(const char*, snort::Value&, snort::SnortConfig*)':
/scratch/union/build_dir/target-x86_64_musl/snort3-3.1.72.0/src/detection/detection_module.cc:259:88: warning: format '%ld' expects argument of type 'long int', but argument 3 has type 'rlim_t' {aka 'long long unsigned int'} [-Wformat=]
  259 |             LogMessage("Capping pcre_match_limit_recursion to %ld, thread stack_size %ld.\n",
      |                                                                                      ~~^
      |                                                                                        |
      |                                                                                        long int
      |                                                                                      %lld
  260 |                 sc->pcre_match_limit_recursion, thread_stack_size);
      |                                                 ~~~~~~~~~~~~~~~~~                       
      |                                                 |
      |                                                 rlim_t {aka long long unsigned int}
[84/986] Building CXX object src/filters/CMakeFiles/filter.dir/detection_filter.cc.o
FAILED: src/filters/CMakeFiles/filter.dir/detection_filter.cc.o 
/scratch/union/staging_dir/toolchain-x86_64_gcc-12.3.0_musl/bin/x86_64-openwrt-linux-musl-g++ -DHAVE_CONFIG_H -Dinline=inline -Drestrict=__restrict -I/scratch/union/build_dir/target-x86_64_musl/snort3-3.1.72.0/src/network_inspectors -I/scratch/union/build_dir/target-x86_64_musl/snort3-3.1.72.0/src -I/scratch/union/staging_dir/target-x86_64_musl/usr/include/luajit-2.1 -I/scratch/union/build_dir/target-x86_64_musl/snort3-3.1.72.0 -I/scratch/union/staging_dir/host/include -I/scratch/union/staging_dir/target-x86_64_musl/usr/include/hs -I/scratch/union/staging_dir/target-x86_64_musl/usr/include/uuid -march=znver3 -mtune=znver3 -O2 -pipe -fno-caller-saves -fno-plt -fhonour-copts -fmacro-prefix-map=/scratch/union/build_dir/target-x86_64_musl/snort3-3.1.72.0=snort3-3.1.72.0 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -I/scratch/union/staging_dir/target-x86_64_musl/usr/include/daq3 -I/scratch/union/staging_dir/target-x86_64_musl/usr/include/tirpc  -fvisibility=hidden   -DNDEBUG  -fno-builtin-malloc -fno-builtin-calloc -fno-builtin-realloc -fno-builtin-free  -DNDEBUG -std=c++14 -MD -MT src/filters/CMakeFiles/filter.dir/detection_filter.cc.o -MF src/filters/CMakeFiles/filter.dir/detection_filter.cc.o.d -o src/filters/CMakeFiles/filter.dir/detection_filter.cc.o -c /scratch/union/build_dir/target-x86_64_musl/snort3-3.1.72.0/src/filters/detection_filter.cc
In file included from /scratch/union/build_dir/target-x86_64_musl/snort3-3.1.72.0/src/filters/detection_filter.cc:26:
/scratch/union/build_dir/target-x86_64_musl/snort3-3.1.72.0/src/hash/xhash.h:63:39: error: 'u_int8_t' has not been declared
   63 |     int release_node(const void* key, u_int8_t type = 0);
      |                                       ^~~~~~~~
[99/986] Building CXX object src/file_api/CMakeFiles/file_api.dir/file_lib.cc.o
ninja: build stopped: subcommand failed.
make[2]: *** [Makefile:147: /scratch/union/build_dir/target-x86_64_musl/snort3-3.1.72.0/.built] Error 1
make[2]: Leaving directory '/scratch/union/feeds/packages/net/snort3'
time: package/feeds/packages/snort3/compile#77.49#10.27#6.34
    ERROR: package/feeds/packages/snort3 failed to build.
make[1]: *** [package/Makefile:120: package/feeds/packages/snort3/compile] Error 1
make[1]: Leaving directory '/scratch/union'
make: *** [/scratch/union/include/toplevel.mk:232: package/snort3/compile] Error 2
```

**Comment by InverseRE** [comment]
Typo in the code. A type name from "sys/types.h".
Definitely, a standard type name is preferable (uint8_t).

We will address that.

**Comment by InverseRE** [comment]
Will be fixed in 3.1.73.0.

**Comment by graysky2** [comment]
Confirmed fixed in 3.1.73.0, ty.
